docs/api-reference/geo-layers/imagery-layer.md

```diff
@@ -62,19 +62,30 @@ new deck._ImageryLayer({});
 
 ## Image Sources
 
-The `ImageryLayer` can accept URLs to geospatial image services such as WMS, however it is also possible to specify a custom URL template. 
+The `ImageryLayer` needs a URL to an image source from which it can start loading map images. The `ImageryLayer` knows how to build URLs for geospatial image services such as WMS. 
+
+However, it is also possible to connect the ImageryLayer to any other REST based service that can render map images from a set of web mercator bounds and a given pixel resolution (perhaps an ArcGIS image server) by specify a custom URL template.
+
+Note that additional features, such as metadata loading, is only supported for known image services, which currently only includes WMS.
 
 ### Layers
 
-Image servers such as WMS can render different layers. Typically as list of layers **must** be specified, otherwise requests for map images will fail.
+Image servers such as WMS can render different layers. Typically as list of layers **must** be specified, otherwise requests for map images will fail. For WMS services, this is controlled by `props.layer`. For other services, layers (if required by that service) can be specified in the template URL, either as a parameter or as a hard-coded part of the template string. 
 
 ### Image Service Metadata
 
-Image services like WMS can often provide metadata (aka capabilities) about the service, listing attribution information, available layers and additional capabilities. The `ImageryLayer` will automatically attempt to query metadata for known service types. Template URLs are specific to image requests and to not support metadata queries.
+Image services like WMS can often provide metadata (aka capabilities) about the service, listing;
+- attribution information, 
+- available layers
+- additional capabilities (pixel/neighborhood queries, legend generation etc). 
+
+The `ImageryLayer` will automatically attempt to query metadata for known service types (currently WMS). 
+
+Template URLs only cover image requests and there is no support for providing a custom URL for the metadata queries. This needs to be handled by the application for non-WMS services.
 
 ### Interactivity
 
-WMS services sometimes provide a mechanism to query a specific pixel. This is supported through a method on the `ImageryLayer`
+WMS services sometimes provide a mechanism to query a specific pixel. This is supported through a `getFeatreuInfo()` method on the `ImageryLayer`
```

Comment:
Is the user supposed to call the method themselves? Document the method?

Reply:
Yes. It is pretty experimental but I feel it won't hurt to document it given that the layer is clearly marked as experimental. It does show that there is more to the WMS protocol than knowing how to hack a TileLayer template URLs and can get users to start playing with interactivity.

Happy to make some additional adjustments, but I don't want to spend too much time bike shedding over this right now as the support will evolve naturally in loaders (for instance there is a version of this that returns actual features rather than just text). If we feel uncomfortable with it here, fine to drop and not mention now, and document it once it has matured.

Reply:
documented the method. Not quite sure what the conventions for layer methods are.